src/keys/mod.rs

```diff
@@ -50,6 +52,17 @@ impl Key {
             Err(_) => false,
         }
     }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
```

Comment:
De-serialization + Serialization allows access to private properties. Do we REALLY need it here?

Reply:
No, we don't. I'll change this.